end-to-end-tests/src/test/java/tech/pegasys/peeps/consensus/qbft/QbftSmartContractValidatorTest.java

```diff
@@ -0,0 +1,129 @@
+  private Web3Provider charlieNode;
+  private Web3Provider deltaNode;
+  private final SignerConfiguration signer = FixedSignerConfigs.ALPHA;
+
+  @Override
+  protected void setUpNetwork(final Network network) {
+    alphaNode = network.addNode("alpha", KeyPair.random());
+    bravoNode = network.addNode("bravo", KeyPair.random());
+    charlieNode = network.addNode("charlie", KeyPair.random());
+    deltaNode = network.addNode("delta", KeyPair.random());
+
+    network.set(
+        ConsensusMechanism.QBFT_SMART_CONTRACT, alphaNode, bravoNode, charlieNode, deltaNode);
+    network.addSigner(signer.name(), signer.resources(), alphaNode);
+  }
+
+  @Test
+  public void consensusAfterSmartContractTransitionMustHappen() throws Exception {
+
+    final List<String> initialValidators =
+        List.of(
+            alphaNode.address().toHexString(),
+            bravoNode.address().toHexString(),
+            charlieNode.address().toHexString(),
+            deltaNode.address().toHexString());
+
+    assertThat(alphaNode.rpc().getBlockNumber()).isLessThan(18);
+
+    ValidatorSmartContractAllowList allowList =
+        ValidatorSmartContractAllowList.deploy(
+                alphaNode.getWeb3j(),
+                signer.getCredentials(),
+                new StaticGasProvider(ZERO, GAS_LIMIT),
+                initialValidators)
+            .send();
+
+    // Note: the transition is configured in the genesis file
+    assertThat(allowList.getContractAddress())
+        .isEqualTo("0xb9a219631aed55ebc3d998f17c3840b7ec39c0cc");
+    assertThat(alphaNode.rpc().getBlockNumber()).isLessThan(20);
+
+    verify().consensusOnBlockNumberIsAtLeast(21);
+
+    allowList
+        .activate(
+            AddressConverter.fromPublicKey(KeyPair.random().publicKey().toHexString())
+                .toHexString())
+        .send();
+    allowList
+        .activate(
+            AddressConverter.fromPublicKey(KeyPair.random().publicKey().toHexString())
+                .toHexString())
+        .send();
+
+    assertThat(allowList.getValidators().send().size()).isEqualTo(6);
+
+    alphaNode.stop();
+    bravoNode.stop();
+
+    final List<Web3Provider> runningNodes = List.of(charlieNode, deltaNode);
+
+    // 2 nodes are up 2 nodes are down and 2 nodes don't exist
+    runningNodes.forEach(this::verifyChainStalled);
+
+    final long stalledBlockNumber = charlieNode.rpc().getBlockNumber();
+
+    alphaNode.start();
+    bravoNode.start();
+
+    final List<Web3Provider> allNodes = List.of(alphaNode, bravoNode, charlieNode, deltaNode);
+    allNodes.forEach(node -> node.awaitConnectivity(allNodes));
+
+    verify().consensusOnBlockNumberIsAtLeast(stalledBlockNumber + 1);
+  }
+
+  private void verifyChainStalled(final Web3Provider web3Provider) {
+    Await.await(
+        () -> {
+          final long startBlockNumber = web3Provider.rpc().getBlockNumber();
+          Thread.sleep(Duration.of(BftConfig.DEFAULT_BLOCK_PERIOD_SECONDS * 2, SECONDS).toMillis());
```

Comment:
Why not just use awaitility with the blockPeriod *2 as the poll time instead the thread sleep?

Reply:
Not sure what you mean. You want to be sure that the block period has at least past and the no new blocks have been produced.